src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -0,0 +1,367 @@
+     *
+     * @param element the element
+     * @return the index
+     */
+    @Override
+    public int indexOf(final Object element) {
+        if (element == null) {
+            for (int i = 0; i < this.index; i++) {
+                if (element == this.data[i]) {
+                    return i;
+                }
+            }
+        } else {
+            for (int i = 0; i < this.index; i++) {
+                if (element.equals(this.data[i])) {
+                    return i;
+                }
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * @param element the element.
+     * @return the index of the last occurrence of the specified element in this
+     *         list, or -1 if this list does not contain the element.
+     */
+    @Override
+    public int lastIndexOf(final Object element) {
+        if (element == null) {
+            for (int i = this.index - 1; i >= 0; i--) {
+                if (element == this.data[i]) {
+                    return i;
+                }
+            }
+        } else {
+            for (int i = this.index - 1; i >= 0; i--) {
+                if (this.data[i].equals(element)) {
+                    return i;
+                }
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Checks if list empty.
+     *
+     * @return true if this list contains no elements.
+     */
+    @Override
+    public boolean isEmpty() {
+        return this.index == 0;
+    }
+
+    /**
+     * Replaces the element at the specified position in this list with the
+     * specified element.
+     *
+     * @param index   the index.
+     * @param element the element.
+     * @return the element previously at the specified position.
+     */
+    @Override
+    public T set(final int index, final T element) {
+        this.checkIndex(index);
+        final T prevElment = get(index);
+        this.data[index] = element;
+        return prevElment;
+    }
+
+    /**
+     * Returns the number of elements in this list.
+     */
+    @Override
+    public int size() {
+        return this.index;
+    }
+
+    /**
+     * Returns a view of the portion of this list between the specified
+     * fromIndex, inclusive, and toIndex, exclusive.
+     *
+     * @param fromIndex the from index.
+     * @param toIndex   the to index.
+     * @return list of specified range within this list.
+     */
+    @Override
+    public List<T> subList(final int fromIndex, final int toIndex) {
+        if (fromIndex < 0 || toIndex > this.index) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        if (fromIndex > toIndex) {
+            throw new IllegalArgumentException(
+                    "fromIndex: " + fromIndex + " > toIndex: " + toIndex);
+        }
+
+        if (fromIndex == toIndex) {
+            return new MyArrayList<T>();
+        }
+
+        final List<T> subList = new MyArrayList<T>();
+
+        for (int i = 0; i < this.index; i++) {
```

Comment:
I gave u comment on this and you marked as resolved.

If list size = million and u need to copy from = 10 to = 20.

Do you need to loop from 0 to Million ??!

Don't resolve comment please without addressing it.

Reply:
I did by adding if condition : 
 if (i > toIndex) {
                break;
            }
so if  from = 10 to = 20. then no need to continue to Million

Reply:
Exactly, if i = 999998 and to 9999999

so u need to loop 999,998 times to get range size 1 (9999999 - 999998).

You can enhance the algorithm better here

Reply:
I am thinking of:
Divide the length of array into 2, then check the "fromIndex" & "lastIndex" if it is in which half of the array. if in first half then divide the first half  and make the same condition.. till we now the range is in which half of half then loop.

Reply:
What ??!!!!

Reply:
I am trying to interpret your weird response :-



Divide the length of array into 2, then check the "fromIndex" & "lastIndex" if it is in which half of the array. 

Why do you need to check in which half??

If list size  = 100
From = 5 and to = 10

So it is in first half, is this information helper you ??!!!

You now have list 1--> 50
Are u trying divide again ??!!!
1--25
1---12

Then what are you trying to do after that ??!!

Reply:
l loop from fromIndex to toIndex